galsim/correlatednoise.py

```diff
@@ -1491,17 +1525,21 @@ def __init__(self, variance, rng=None, scale=None, wcs=None, gsparams=None):
             if not isinstance(wcs, galsim.BaseWCS):
                 raise TypeError("wcs must be a BaseWCS instance")
             if not wcs.isUniform():
-                raise ValueError("Cannot provide non-local wcs")
+                raise ValueError("Cannot provide non-uniform wcs")
         elif scale is not None:
             wcs = galsim.PixelScale(scale)
         else:
             wcs = galsim.PixelScale(1.0)
 
-        # Need variance == xvalue(0,0)
-        # Pixel has flux of f/scale^2, so use f = variance * scale^2
-        image_pix = galsim.Pixel(scale=1.0, flux=variance * wcs.pixelArea(), gsparams=gsparams)
-        world_pix = wcs.toWorld(image_pix)
-        cf_object = galsim.AutoConvolve(world_pix)
-        _BaseCorrelatedNoise.__init__(self, rng, cf_object)
+        # Need variance == xvalue(0,0) after autoconvolution
+        # So the Pixel needs to have an amplitude of sigma at (0,0)
+        import math
```

Comment:
For non-arrays, `math.sqrt` is indeed faster, since the `numpy` function has to check if the input is an array or not, whereas the `math` version only works for scalar floats.  I just checked, and the difference is about a factor of 3.

But I'm sure the difference is negligible for uses such as this.  My import statement here is really just because I don't generally import things at the top of the .py file as much as Barney does, so I wasn't thinking that numpy might already be imported.  I generally just import what I need in each function and use it there.  Python is smart enough not to waste any time importing things again if it has already loaded a module into memory.  So there's not really any downside to putting the imports in the functions, and I find it a bit easier to keep track of them there.